e2e/cypress/integration/bot_accounts/bot_api_spec.js

```diff
@@ -0,0 +1,295 @@
+                    cy.wrap(el[0].parentElement.parentElement).find('.item-actions > button').click();
+                });
+
+                // # Try to post again
+
+                // * Validate that posting works
+                cy.postBotMessage({channelId: channel.id, message: msg2, token});
+
+                // * Validate post presence
+                cy.visit(`/${newTeam.name}/channels/` + channel.name);
+
+                cy.getLastPostId().then((postId) => {
+                    // * Validate post was created
+                    cy.get(`#postMessageText_${postId}`, {timeout: TIMEOUTS.ONE_MIN}).should('contain', msg2);
+                });
+            });
+        });
+    });
+
+    it('MM-T1878 Disable token can not be used to post', () => {
+        // # Create private channel that bot doesn't belong to
+        cy.apiCreateDirectChannel([newUser.id, adminUser.id]).then(({channel}) => {
+            // # Create token for the bot
+            cy.apiAccessToken(newBot.user_id, 'some text').then(({token, id}) => {
+                const msg1 = 'this is a bot message ' + botName;
+
+                // # Create a post
+                cy.postBotMessage({channelId: channel.id, message: msg1, token});
+
+                // # Validate post presence
+                cy.visit(`/${newTeam.name}/channels/` + channel.name);
+
+                cy.getLastPostId().then((postId) => {
+                    // * Validate post was created
+                    cy.get(`#postMessageText_${postId}`, {timeout: TIMEOUTS.ONE_MIN}).should('contain', msg1);
+                });
+
+                // # Disable the bot token
+                cy.visit(`/${newTeam.name}/integrations/bots`);
+
+                cy.findByText(`Test Bot (@${botName})`).then((el) => {
+                    // # Make sure it's on the screen
+                    cy.wrap(el[0].parentElement.parentElement).scrollIntoView();
+                    cy.get(`[name="${id}_deactivate"]`).click();
+                });
+
+                // # Try to post again
+                const msg2 = 'this is a bot message2 ' + botName;
+
+                // # Create a post
+                cy.postBotMessage({channelId: channel.id, message: msg2, token, failOnStatus: false}).then(({status}) => {
+                    // * Validate that posting failed
+                    expect(status, 403);
+                });
+
+                // # Enable the bot token again
+                cy.visit(`/${newTeam.name}/integrations/bots`);
+
+                cy.findByText(`Test Bot (@${botName})`).then((el) => {
+                    // # Make sure it's on the screen
+                    cy.wrap(el[0].parentElement.parentElement).scrollIntoView();
+                    cy.get(`[name="${id}_activate"]`).click();
+                });
+
+                // # Try to post again
+                // * Validate that posting works
+                cy.postBotMessage({channelId: channel.id, message: msg2, token});
```

Comment:
Per test case, this should fail ("On 6. Post fails") but I don't know if that's correct. I'm clarifying that to our QA team.

Reply:
Dylan confirmed this and the one you did is correct. Test case is updated and is now `On 6. Post succeeds`.